src/server.rs

```diff
@@ -684,4 +690,45 @@ impl WalletService for Arc<crate::wallet::Wallet> {
             "not implemented",
         ))
     }
+
+    async fn ripemd160(
+        &self,
+        request: tonic::Request<Ripemd160Request>,
+    ) -> std::result::Result<tonic::Response<Ripemd160Response>, tonic::Status> {
+        let message = request.into_inner().message;
+        let message: Vec<u8> = message
+            .ok_or_else(|| missing_field::<GetBmmHStarCommitmentRequest>("message"))?
+            .decode_tonic::<GetBlockHeaderInfoRequest, _>("message")?;
+        let mut hasher = ripemd::Ripemd160::new();
+        hasher.update(&message);
+        let hash = hasher.finalize();
+        let response = Ripemd160Response {
+            hash: Some(ConsensusHex::encode(&hash.as_slice().to_vec())),
+        };
+        Ok(tonic::Response::new(response))
+    }
+
+    async fn hmac_sha512(
+        &self,
+        request: tonic::Request<HmacSha512Request>,
+    ) -> std::result::Result<tonic::Response<HmacSha512Response>, tonic::Status> {
+        let HmacSha512Request { secret, message } = request.into_inner();
+        let secret: Vec<u8> = secret
+            .ok_or_else(|| missing_field::<GetBmmHStarCommitmentRequest>("secret"))?
+            .decode_tonic::<GetBlockHeaderInfoRequest, _>("secret")?;
+        let message: Vec<u8> = message
+            .ok_or_else(|| missing_field::<GetBmmHStarCommitmentRequest>("message"))?
+            .decode_tonic::<GetBlockHeaderInfoRequest, _>("message")?;
```

Comment:
This is a mistake, you should use `decode_tonic::<HmacSha512Request>("message")`